nextcore/gateway/shard_manager.py

```diff
@@ -188,17 +189,20 @@ async def connect(self) -> None:
         else:
             shard_ids = self.shard_ids
 
-        for shard_id in shard_ids:
-            shard = self._spawn_shard(shard_id, self._active_shard_count)
+        async with create_task_group() as tg:
+            for shard_id in shard_ids:
+                shard = self._spawn_shard(shard_id, self._active_shard_count)
+                # Here we lazy connect the shard. This gives us a bit more speed when connecting large sets of shards.
+                await tg.spawn(shard.connect)
```

Comment:
Doesnt this wait?

Reply:
Right so this will work, however start_soon should be used instead.

Reply:
`TaskGroup.spawn` is used in the latest commit. I realized it was deprecated after running the unit tests.